### PR TITLE
Cleans up shower code

### LINKED ISSUE #41773: Monkey cubes do not expand in a running shower
Round ID:  97984

1) Take monkey cube.
2) Turn on shower.
3) Throw monkey cube in running shower.
4) No response.

IT should instead grow into a monkey.

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
fix: Monkey cubes now expand in a running shower!
fix: Slimes now die in running showers.
/:cl:

Showers now apply 200 water reagent via TOUCH every tick to anything under them.
This doesn't actually add reagent to the atoms it touches though.
Showers now use timers rather than spawn for the mist creation.
Mist isn't recreated every time the shower is turned on and off.
All living mobs are burned by hot showers, not just carbons.
This means that slimes are healed by hot showers.
Showers are now in a separate file.
Shower temperatures now use defines.

Fixes #41773.

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -1. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by coiax** [comment]
THIS IS CODE IMPROVEMENT AND REFACTOR, LEAVE ME ALONE

File: code/game/objects/structures/shower.dm [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	if(I.use_tool(src, user, 50))
+		switch(current_temperature)
+			if(SHOWER_NORMAL)
+				current_temperature = SHOWER_FREEZING
+			if(SHOWER_FREEZING)
+				current_temperature = SHOWER_BOILING
+			if(SHOWER_BOILING)
+				current_temperature = SHOWER_NORMAL
+		user.visible_message("<span class='notice'>[user] adjusts the shower with \the [I].</span>", "<span class='notice'>You adjust the shower with \the [I] to [current_temperature] temperature.</span>")
+		log_game("[key_name(user)] has wrenched a shower to [current_temperature] at ([x],[y],[z])")
```
Comment (ShizCalev) [inline code/game/objects/structures/shower.dm]: move this to the individual's combat log instead. also switch the XYZ at the end to use either COORD or AREACOORD instead.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by nicbn** [comment]
Could you please make the temperature be a number representing the actual temperature instead of text defines?

**Comment by ExcessiveUseOfCobblestone** [comment]
If you used the temp instead of strings you could save some lines anon

**Comment by coiax** [comment]
But then you'd need to turn the temperature back into "scalding", "freezing"

**Comment by ExcessiveUseOfCobblestone** [comment]
nani

File: code/game/objects/structures/shower.dm [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	on = !on
+	update_icon()
+	handle_mist()
+	add_fingerprint(M)
+	if(on)
+		START_PROCESSING(SSmachines, src)
+		process()
+		soundloop.start()
+		wash_atom(loc)
+		for(var/atom/movable/G in loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/shower.dm]:
```suggestion
 		for(var/G in loc)
```

File: code/game/objects/structures/shower.dm [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+/obj/machinery/shower/proc/wash_mob(mob/living/L)
+	SEND_SIGNAL(L, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+	L.wash_cream()
+	L.remove_atom_colour(WASHABLE_COLOUR_PRIORITY)
+	SEND_SIGNAL(L, COMSIG_ADD_MOOD_EVENT, "shower", /datum/mood_event/nice_shower)
+	if(iscarbon(L))
+		var/mob/living/carbon/M = L
+		. = TRUE
+		for(var/obj/item/I in M.held_items)
```
Comment (Cyberboss) [inline code/game/objects/structures/shower.dm]:
```suggestion
 		for(var/I in M.held_items)
```

File: code/game/objects/structures/shower.dm [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	if(strength <= RAD_BACKGROUND_RADIATION)
+		qdel(healthy_green_glow)
+		return
+	healthy_green_glow.strength -= max(0, (healthy_green_glow.strength - (RAD_BACKGROUND_RADIATION * 2)) * 0.2)
+
+/obj/machinery/shower/process()
+	if(on)
+		wash_atom(loc)
+		for(var/atom/movable/AM in loc)
+			wash_atom(AM)
```
Comment (Cyberboss) [inline code/game/objects/structures/shower.dm]: Duplicated set of lines

File: code/game/objects/structures/shower.dm [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	else
+		return PROCESS_KILL
+
+/obj/machinery/shower/deconstruct(disassembled = TRUE)
+	new /obj/item/stack/sheet/metal(loc, 3)
```
Comment (Cyberboss) [inline code/game/objects/structures/shower.dm]:
```suggestion
 	new /obj/item/stack/sheet/metal(drop_location(), 3)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
awaiting review resolving

File: code/game/objects/structures/shower.dm (hunk 4 above)
Comment (coiax) [reply code/game/objects/structures/shower.dm]: You talking about the duplication with when it's turned on in interact() ?

**Comment by SpaceManiac** [comment]
Conflict with #41986

File: code/game/objects/structures/shower.dm (hunk 4 above)
Comment (Cyberboss) [reply code/game/objects/structures/shower.dm]: Yes

Comment (coiax) [reply code/game/objects/structures/shower.dm]: Oh, I see what you mean now. My bad.

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/structures/shower.dm]: is this fixed

**Pull request merged**